tests/unit/utils/test_os_utils.py

```diff
@@ -15,12 +15,36 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 import os
+from pathlib import Path
 
 import pytest
 
 from craft_parts.utils import os_utils
 
 
+class TestTimedWriter:
+    """Check if minimum interval ensured between writes."""
+
+    @pytest.mark.usefixtures("new_dir")
```

Comment:
these should ideally be 3 distinct tests, unless for organizational reasons, these don't need to be part of a class either.

Reply:
Split into three tests, kept class for organization.